File: win-prog/ProgWinVB/An Exercise in Text Output/SysInfoList/SysInfoList.vb
'--------------------------------------------
' SysInfoList.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class SysInfoList
    Inherits Form

    ReadOnly cxCol As Single
    ReadOnly cySpace As Integer

    Shared Sub Main()
        Application.Run(New SysInfoList())
    End Sub

    Sub New()
        Text = "System Information: List"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText

        Dim grfx As Graphics = CreateGraphics()
        Dim szf As SizeF = grfx.MeasureString(" ", Font)
        cxCol = szf.Width + SysInfoStrings.MaxLabelWidth(grfx, Font)
        grfx.Dispose()

        cySpace = Font.Height
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim br As New SolidBrush(ForeColor)
        Dim iCount As Integer = SysInfoStrings.Count
        Dim astrLabels() As String = SysInfoStrings.Labels
        Dim astrValues() As String = SysInfoStrings.Values
        Dim i As Integer

        For i = 0 To iCount - 1
            grfx.DrawString(astrLabels(i), Font, br, 0, i * cySpace)
            grfx.DrawString(astrValues(i), Font, br, cxCol, i * cySpace)
        Next i
    End Sub
End Class

File: win-prog/ProgWinCS/An Exercise in Text Output/SysInfoList/SysInfoList.cs
//------------------------------------------
// SysInfoList.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class SysInfoList: Form
{
     readonly float cxCol;
     readonly int   cySpace;

     public static void Main()
     {
          Application.Run(new SysInfoList());
     }
     public SysInfoList()
     {
          Text = "System Information: List";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;

          Graphics grfx  = CreateGraphics();
          SizeF    sizef = grfx.MeasureString(" ", Font);
          cxCol = sizef.Width + SysInfoStrings.MaxLabelWidth(grfx, Font);
          grfx.Dispose();
          cySpace = Font.Height;
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics grfx       = pea.Graphics;
          Brush    brush      = new SolidBrush(ForeColor);
          int      iCount     = SysInfoStrings.Count;
          string[] astrLabels = SysInfoStrings.Labels;
          string[] astrValues = SysInfoStrings.Values;

          for (int i = 0; i < iCount; i++)
          {
               grfx.DrawString(astrLabels[i], Font, brush, 
                               0, i * cySpace);
               grfx.DrawString(astrValues[i], Font, brush, 
                               cxCol, i * cySpace); 
          }
     }
}
